src/Grpc.Net.Client/Balancer/DnsResolver.cs

```diff
@@ -165,33 +124,13 @@ protected override void Dispose(bool disposing)
             base.Dispose(disposing);
 
             _timer?.Dispose();
-            _cts.Cancel();
-            _listener = null;
-            _disposed = true;
         }
 
-        private async void OnTimerCallback(object? state)
+        private void OnTimerCallback(object? state)
         {
             try
             {
-                var awaitRefresh = false;
-                lock (_lock)
-                {
-                    if (_refreshTask.IsCompleted)
-                    {
-                        _refreshTask = RefreshCoreAsync(_cts.Token);
-                        awaitRefresh = true;
-                    }
-                }
-
-                if (awaitRefresh)
-                {
-                    await _refreshTask.ConfigureAwait(false);
-                }
-            }
-            catch (OperationCanceledException)
```

Comment:
Why is it safe to remove the catch for the `OperationCanceledException` here?

Reply:
Refresh will never throw. However, there is a new try/catch in the base resolver that should probably ignore cancellation. I will update that code.